test/test_transforms_v2_refactored.py

```diff
@@ -0,0 +1,679 @@
+            antialias=True,
+        )
+
+    @pytest.mark.parametrize("size", OUTPUT_SIZES)
+    # `InterpolationMode.NEAREST` is modeled after the buggy `INTER_NEAREST` interpolation of CV2.
+    # The PIL equivalent of `InterpolationMode.NEAREST` is `InterpolationMode.NEAREST_EXACT`
+    @pytest.mark.parametrize("interpolation", set(INTERPOLATION_MODES) - {transforms.InterpolationMode.NEAREST})
+    @pytest.mark.parametrize("use_max_size", [True, False])
+    def test_kernel_image_tensor_correctness(self, size, interpolation, use_max_size):
+        if not (max_size_kwarg := self._make_max_size_kwarg(use_max_size=use_max_size, size=size)):
+            return
+
+        image = self._make_input(torch.Tensor, dtype=torch.uint8, device="cpu")
+
+        actual = F.resize_image_tensor(
+            image,
+            size=size,
+            interpolation=interpolation,
+            **max_size_kwarg,
+            # antialias is always True for PIL
+            antialias=True,
+        )
+        expected = F.to_image_tensor(
+            F.resize_image_pil(
+                F.to_image_pil(image),
+                size=size,
+                interpolation=interpolation,
+                **max_size_kwarg,
+            )
+        )
+
+        self._check_size(image, actual, size=size, **max_size_kwarg)
+
+        mae = (actual.float() - expected.float()).abs().mean()
+        assert mae < 1
+
+    @pytest.mark.parametrize("size", OUTPUT_SIZES)
+    # `InterpolationMode.NEAREST` is modeled after the buggy `INTER_NEAREST` interpolation of CV2.
+    # The PIL equivalent of `InterpolationMode.NEAREST` is `InterpolationMode.NEAREST_EXACT`
+    @pytest.mark.parametrize("interpolation", set(INTERPOLATION_MODES) - {transforms.InterpolationMode.NEAREST})
+    @pytest.mark.parametrize("use_max_size", [True, False])
+    def test_transform_image_correctness(self, size, interpolation, use_max_size):
+        if not (max_size_kwarg := self._make_max_size_kwarg(use_max_size=use_max_size, size=size)):
+            return
+
+        transform = transforms.Resize(size=size, interpolation=interpolation, **max_size_kwarg, antialias=True)
+
+        image = self._make_input(torch.Tensor, dtype=torch.uint8, device="cpu")
+
+        actual = transform(image)
+        expected = F.to_image_tensor(
+            F.resize_image_pil(F.to_image_pil(image), size=size, interpolation=interpolation, **max_size_kwarg)
+        )
+
+        self._check_size(image, actual, size=size, **max_size_kwarg)
+
+        mae = (actual.float() - expected.float()).abs().mean()
+        assert mae < 1
+
+    @pytest.mark.parametrize("interpolation", set(transforms.InterpolationMode) - set(INTERPOLATION_MODES))
+    def test_pil_interpolation_compat_smoke(self, interpolation):
+        F.resize_image_pil(
+            self._make_input(PIL.Image.Image, dtype=torch.uint8, device="cpu"),
+            size=self.OUTPUT_SIZES[0],
+            interpolation=interpolation,
+        )
```

Comment:
Might be worth checking that we raise a proper error when trying those modes on tensors? I assume this is a pre-existing test, so feel free to leave it out.